support/k8s/k8s-workload-registrar/informerhandler.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+type InformerHandlerConfig struct {
+	Log        logrus.FieldLogger
+	Controller *Controller
+	Factory    informers.SharedInformerFactory
+}
+
+type InformerHandler struct {
+	c InformerHandlerConfig
+}
+
+func NewInformerHandler(config InformerHandlerConfig) *InformerHandler {
+	return &InformerHandler{
+		c: config,
+	}
+}
+
+func (ih *InformerHandler) Run(ctx context.Context) error {
+	podInformer := ih.c.Factory.Core().V1().Pods().Informer()
+	podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*corev1.Pod)
+			log := ih.c.Log.WithFields(logrus.Fields{
+				"ns":  pod.Namespace,
+				"pod": pod.Name,
+			})
+			if err := ih.c.Controller.SyncPod(ctx, pod); err != nil {
+				log.WithError(err).Error("Failed to sync pod")
```

Comment:
If this fails, will it ever try again, or will it be broken forever?

Reply:
(After latest changes)

The informer refetches everything from the apiserver at regular resync intervals, controlled by the informer_resync_interval flag in the config file.

This appears as a new add or update event for everything that has already been seen, so will end up calling SyncPod on everything again.

Off by default, as this can be quite expensive on the system if the cluster has many pods, so there is no safe default interval.